### ISSUE TITLE
Adding support for displaying more than 3 numbers at a time

### ISSUE THREAD

**Issue by alexbhandari** [MarcinusX/NumberPicker#68]
I am looking into displaying more than 3 numbers and have not found any solutions for this. I started seeing if I could do this by making modifications to the numberpicker class. I am exploring specifically for horizontal, but I think it would work both ways. 

So far I have managed this by allowing list view width to be passed into the constructor so setting `listViewWidth = 7 * itemExtent,` gives this

![image](https://user-images.githubusercontent.com/3032338/76669209-b23b9c80-6558-11ea-84cf-18f75346cf3e.png)

As others have noted (#60 and [on stackoverflow](https://stackoverflow.com/questions/56723091/how-to-show-more-element-of-number-picker-list-in-flutter)) this results in some issues. These issues seem to be coming from the `intValueInTheMiddle` and other logic for highlighting the middle number and also preventing the max number from being highlighted. 
```
  bool _onIntegerNotification(Notification notification) {
    if (notification is ScrollNotification) {
      //calculate
      int intIndexOfMiddleElement =
      (notification.metrics.pixels / itemExtent).round();
      if (!infiniteLoop) {
        intIndexOfMiddleElement =
            intIndexOfMiddleElement.clamp(0, integerItemCount - 1);
      }
      int intValueInTheMiddle = _intValueFromIndex(intIndexOfMiddleElement + 1);
      intValueInTheMiddle = _normalizeIntegerMiddleValue(intValueInTheMiddle);
```
I am going to be looking into this some more, but wanted to post this here in case anyone had any pointers or thoughts on this approach. It seems to me that this should be doable as it is a matter of tweaking the `intValueInTheMiddle` and selection logic to work with an adaptable length.

Below is my tweak to the constructor for increasing past 3 numbers as shown above:
```
  NumberPicker.horizontal({
        Key key,
        @required int initialValue,
        @required this.minValue,
        @required this.maxValue,
        @required this.onChanged,
        this.textMapper,
        this.itemExtent = kDefaultItemExtent,
        this.listViewHeight = kDefaultListViewCrossAxisSize,
        double listViewWidth,
        this.step = 1,
        this.zeroPad = false,
        this.highlightSelectedValue = true,
        this.decoration,
  })  : assert(initialValue != null),
                assert(minValue != null),
                assert(maxValue != null),
                assert(maxValue > minValue),
                assert(initialValue >= minValue && initialValue <= maxValue),
                assert(step > 0),
                selectedIntValue = initialValue,
                selectedDecimalValue = -1,
                decimalPlaces = 0,
                intScrollController = new ScrollController(
                  initialScrollOffset: (initialValue - minValue) ~/ step * itemExtent,
                ),
                scrollDirection = Axis.horizontal,
                decimalScrollController = null,
                listViewWidth = listViewWidth ?? 3 * itemExtent,
                infiniteLoop = false,
                integerItemCount = (maxValue - minValue) ~/ step + 1,
                super(key: key);
```

**Comment by alexbhandari** [comment]
Managed to get this working. Should work for an arbitrary odd display number of 3 or greater. Still needs some testing, I have only checked the horizontal version currently, but code is here: https://github.com/alexbhandari/NumberPicker/commit/4535e35269f88b3ba331ce7249a7abd3bbcfb2fc

here's a screenshot using 7 display numbers:
![image](https://user-images.githubusercontent.com/3032338/76672549-8a096900-656b-11ea-9d13-d1f4fd440e26.png)
![image](https://user-images.githubusercontent.com/3032338/76672554-9392d100-656b-11ea-8ac3-63635a79e525.png)

5 display numbers:
![image](https://user-images.githubusercontent.com/3032338/76672564-9f7e9300-656b-11ea-82c4-7387b6586726.png)

and works for the default case of 3 as well:
![image](https://user-images.githubusercontent.com/3032338/76672571-b02f0900-656b-11ea-8350-d6dbd45e88b8.png)

**Comment by lzhuor** [comment]
Hi @alexbhandari , thanks for your work! I tested it for Integer picker and decimal picker. Fixed `listViewHeight` in their constructors. 

Created a PR #73.

**Comment by alexbhandari** [comment]
@lzhuor awesome! thanks for making that fix and the PR. I'm excited to see this merged in

**Comment by MarcinusX** [comment]
I think it's fixed in #102. Feel free to reopen if disagree. :)
